src/modules/keyboardmanager/ui/KeyDropDownControl.h

```diff
@@ -0,0 +1,190 @@
+#pragma once
+#include <keyboardmanager/common/KeyboardManagerState.h>
+
+// Wrapper class for the key drop down menu
+class KeyDropDownControl
+{
+private:
+    // Stores the drop down combo box
+    ComboBox dropDown;
+    // Stores the previous layout
+    HKL previousLayout = 0;
+    // Stores the key code list
+    std::vector<DWORD> keyCodeList;
+
+    // Function to set properties apart from the SelectionChanged event handler
+    void SetDefaultProperties(bool isShortcut);
+
+    // Function to check if the layout has changed and accordingly update the drop down list
+    void CheckAndUpdateKeyboardLayout(ComboBox currentDropDown, bool isShortcut);
+
+public:
+    // Pointer to the keyboard manager state
+    static KeyboardManagerState* keyboardManagerState;
+
+    // Constructor for single key drop down
+    KeyDropDownControl(size_t rowIndex, size_t colIndex, std::vector<std::vector<DWORD>>& singleKeyRemapBuffer)
+    {
+        SetDefaultProperties(false);
+        dropDown.SelectionChanged([&, rowIndex, colIndex](IInspectable const& sender, SelectionChangedEventArgs const& args) {
+            ComboBox currentDropDown = sender.as<ComboBox>();
+            int selectedKeyIndex = currentDropDown.SelectedIndex();
+
+            // Check if the element was not found or the index exceeds the known keys
+            if (selectedKeyIndex != -1 && keyCodeList.size() > selectedKeyIndex)
+            {
+                singleKeyRemapBuffer[rowIndex][colIndex] = keyCodeList[selectedKeyIndex];
+            }
+            else
+            {
+                // Reset to null if the key is not found
+                singleKeyRemapBuffer[rowIndex][colIndex] = NULL;
+            }
+        });
+    }
+
+    // Constructor for shortcut drop down
+    KeyDropDownControl(size_t rowIndex, size_t colIndex, std::vector<std::vector<Shortcut>>& shortcutRemapBuffer, std::vector<std::unique_ptr<KeyDropDownControl>>& keyDropDownControlObjects, StackPanel parent)
```

Comment:
Maybe moving this to a method would be cleaner.

Reply:
I moved the common part in both the constructors to a separate function. The only thing defined in the two constructors are the selectionchanged handlers.